prefect_aws/workers/ecs_worker.py

```diff
@@ -0,0 +1,1354 @@
+
+        # Ensure configuration tags and env are respected post-templating
+
+        tags = [
+            item
+            for item in task_run_request.get("tags", [])
+            if item["key"] not in configuration.labels.keys()
+        ] + [
+            {"key": k, "value": v}
+            for k, v in configuration.labels.items()
+            if v is not None
+        ]
+        if tags:
+            task_run_request["tags"] = tags
+
+        if orchestration_container:
+            environment = [
+                item
+                for item in orchestration_container.get("environment", [])
+                if item["name"] not in configuration.env.keys()
+            ] + [
+                {"name": k, "value": v}
+                for k, v in configuration.env.items()
+                if v is not None
+            ]
+            if environment:
+                orchestration_container["environment"] = environment
+
+        # Remove empty container overrides
+
+        overrides["containerOverrides"] = [v for v in container_overrides if v]
+
+        return task_run_request
+
+    def _create_task_run(self, ecs_client: _ECSClient, task_run_request: dict) -> str:
+        """
+        Create a run of a task definition.
+
+        Returns the task run ARN.
+        """
+        return ecs_client.run_task(**task_run_request)["tasks"][0]
+
+    def _task_definitions_equal(self, taskdef_1, taskdef_2) -> bool:
+        """
+        Compare two task definitions.
+
+        Since one may come from the AWS API and have populated defaults, we do our best
+        to homogenize the definitions without changing their meaning.
+        """
+        if taskdef_1 == taskdef_2:
+            return True
+
+        if taskdef_1 is None or taskdef_2 is None:
+            return False
+
+        taskdef_1 = copy.deepcopy(taskdef_1)
+        taskdef_2 = copy.deepcopy(taskdef_2)
+
+        def _set_aws_defaults(taskdef):
+            """Set defaults that AWS would set after registration"""
+            container_definitions = taskdef.get("containerDefinitions", [])
+            essential = any(
+                container.get("essential") for container in container_definitions
+            )
+            if not essential:
+                container_definitions[0].setdefault("essential", True)
+
+            taskdef.setdefault("networkMode", "bridge")
+
+        _set_aws_defaults(taskdef_1)
+        _set_aws_defaults(taskdef_2)
+
+        def _drop_empty_keys(dict_):
+            """Recursively drop keys with 'empty' values"""
+            for key, value in tuple(dict_.items()):
+                if not value:
+                    dict_.pop(key)
+                if isinstance(value, dict):
+                    _drop_empty_keys(value)
+                if isinstance(value, list):
+                    for v in value:
+                        if isinstance(v, dict):
+                            _drop_empty_keys(v)
```

Comment:
I started to do this but it's pretty specific to the expected types here. It only handles lists and dicts and would be misleading in general usage.

Reply:
Can you move it elsewhere in this module? It doesn't look like it needs to be declared directly in this module.

Reply:
Can you explain why that would make it better? It should not be used outside of this context and is only implemented as a function because it needs to recurse.

Reply:
My main concerns are readability and instantiating a function within a function that doesn't use any variables from an external closure. These are loosely held opinions so I'm good with whatever you choose 😃

Reply:
I also don't feel strongly, I just don't know if separating it out elsewhere will improve readability and then it may be used in an inappropriate context later. I just copied this from the previous implementation and I would have implemented it inline (i.e. as a for loop) if I did not need a function for recursion.

Reply:
Hopefully https://github.com/PrefectHQ/prefect-aws/pull/238/commits/bc727bc81cad666c9a6ee04368331a8c3e9f12f7 addresses both of our concerns.